flang/lib/Lower/ConvertExpr.cpp

```diff
@@ -2464,9 +2514,10 @@ class ArrayExprLowering {
     auto exv = Fortran::lower::createSomeInitializerExpression(
         loc, converter, toEvExpr(x), symMap, stmtCtx);
     auto val = fir::getBase(exv);
-    mlir::Value mem = builder.create<fir::AllocMemOp>(loc, val.getType());
-    stmtCtx.attachCleanup([=]() { builder.create<fir::FreeMemOp>(loc, mem); });
-    builder.create<fir::StoreOp>(loc, val, mem);
+    auto *bldr = &converter.getFirOpBuilder();
```

Comment:
What is the motivation for getting the builder here?

Reply:
Correctness. We can't forward a copy of a reference sitting on the stack to the future lambda for cleanup.

Reply:
Thanks for the explanation.